javascript/javascript2/week1/main.js

```diff
@@ -0,0 +1,90 @@
+//Find the shortest word
+function notThisFunctionName(parameter) {
+  if (Array.isArray(parameter)) {
+    let word = parameter[0];
+    for (let l = 0; l < parameter.length; l++) {
+      if (word.length > parameter[l].length) {
+        word = parameter[l];
+      }
+    }
+    return "shortest word is '" + word + "'";
+  } else {
+    //Find and count the Danish letters
+    let total = 0;
+    let myList = [];
+    for (let i = 0; i < myArray.length; i++) {
+      let k = 0;
+      for (let j = 0; j < parameter.length; j++) {
+        if (myArray[i] === parameter[j]) {
+          k += 1;
+        }
+      }
+      if (k > 0) {
+        myList.push({ [myArray[i]]: k });
+        total += k;
+      }
+    }
+    myList.unshift({ Total: total });
+    return myList;
+  }
+}
+const danishWords = ["bil", "plante", "kaffe", "bog", "ø", "planetarium"];
+const danishString = "Jeg hør en blå bil";
+const danishString2 = "Blå grød med røde bær";
+const myArray = ["å", "ø", "æ"];
+console.log(notThisFunctionName(danishWords));
+console.log(notThisFunctionName(danishString));
+console.log(notThisFunctionName(danishString2));
+
+//Spirit animal name generator
+const body = document.querySelector("body");
+const myButton = document.querySelector("button");
+myButton.innerText = "click here";
+const spiritAnimalArray = [
+  "The confident bear",
+  "The leading lion",
+  "The strong willed tiger",
+  "The clever fox",
+  "The loyal wolf",
+  "The peaceful panda",
+  "The crying butterfly",
+  "The wise horse",
+  "The compationate deer",
+  "The patient eagle",
+];
+let newSpiritAnimalArray = [...spiritAnimalArray];
+function getInputValue() {
+  var inputValue = document.getElementById("myInput").value;
+  if (inputValue) {
+    const randomAnimal = Math.floor(Math.random() * spiritAnimalArray.length);
+    document.getElementById("displayPara").innerHTML =
+      "<b>Name: </b>" +
+      inputValue +
+      ": " +
+      inputValue +
+      " - " +
+      spiritAnimalArray[randomAnimal];
+    document.getElementById("firstSelection").disabled = true;
+    newSpiritAnimalArray.splice([randomAnimal], 1);
+  } else {
+    document.getElementById("displayPara").innerHTML = " Enter your name first";
+  }
+}
+
+function getNewInputValue() {
+  var inputValue = document.getElementById("myInput").value;
```

Comment:
favor using let/const instead of var, as it is the old way of declaring variables. If you needed some more convincing and wanted to dig deep into it, this is quite cool blog post I found explaining the differences and meaning behind : https://medium.com/@codingsam/awesome-javascript-no-more-var-working-title-999428999994 🦸‍♀️